scipy/signal/_max_len_seq_inner.py

```diff
@@ -0,0 +1,23 @@
+# Author: Eric Larson
+# 2014
+
+import numpy as np
+
+#pythran export _max_len_seq_inner(intc[], int8[], int, int, int8[])
+#pythran export _max_len_seq_inner(intc[::], int8[], int, int, int8[])
```

Comment:
`intc` cannot really be right, because it's platform-dependent. I think using `int32` and `int64` is the right thing to do here.

Reply:
Related question, what does the `(reshaped)` mean in:
```
E   TypeError: Invalid call to pythranized function `_max_len_seq_inner(int64[:] (reshaped), int8[:], int, int, int8[:])'
E   Candidates are:
E   
E       - _max_len_seq_inner(int64[::], int8[:], int, int, int8[:])
E       - _max_len_seq_inner(int64[:], int8[:], int, int, int8[:])
```
the signatures seem to match other than that.

Reply:
I'm not quite sure. It really should be `intp` as in the call site, but I'm trying to figure out why it gets refused.

Reply:
It doesn't seem to like:
```
taps = np.unique(np.array(taps, np.intp))[::-1]
```
the slicing gives it strides `(-8,)` rather than `(8,)`. [This code](https://github.com/serge-sans-paille/pythran/blob/651ebc0e4821454b07173e235b946ddbbfe99f10/pythran/pythonic/python/core.hpp#L87) adds the `(reshaped)`. Some reshaped arrays seem to not be supported, says [this test](https://github.com/serge-sans-paille/pythran/blob/651ebc0e4821454b07173e235b946ddbbfe99f10/pythran/tests/test_conversion.py#L124).

Reply:
The [actual code for the dtype](https://github.com/serge-sans-paille/pythran/blob/1ddfaf7f790c5847d4571c1493dfccb0d5c2c289/pythran/pythonic/include/types/numpy_op_helper.hpp#L183) says:
```
  /* A reshaped type create a type that has the same shape as C && the same
   * dtype as T
   * To the opposite of an adapted type, it does *not* changes constants type
   */
```
That templated code I don't understand though, so I'll stop digging there.